### PR TITLE
[JENKINS-42816] use agent terminology and fix help for label.

### PR DESCRIPTION
**Pull request by aHenryJard**
See [JENKINS-42816](https://issues.jenkins.io/browse/JENKINS-42816).
- use agent terminology
- the help message for label was not displayed, added the missing field value to have it displayed.

I have run tests locally, it seems ok

Screenshot of the help message (using `mvn hpi:run`):
agent terminology

### Your checklist for this pull request

- [x] Pull request title represents the changelog entry which will be used in Release Notes. JIRA issue is a part of the title (see existing PRs as examples)
- [x] Please describe what you did. Short summary for reviewers. If the change involves WebUI, add screenshots 
- [x] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org) or pull requests
- [x] Test automation, if relevant. We expect autotests for all new features or complex bugfixes
- [x] Documentation if relevant (new features). We want to use Documentation-as-Code, just put docs to README or to new Doc files
- [x] Javadoc for new APIs. Any public/protected method is considered as API unless annotated by `Restricted` ([docs](https://kohsuke.org/access-modifier/apidocs/org/kohsuke/accmod/AccessRestriction.html)) 

### CC

@mention

### CODE REVIEW

File: src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly [hunk 1]
```diff
@@ -44,7 +44,7 @@
       <f:optionalBlock name="hasAssignedLabel" title="${%Restrict where this promotion process can be run}"
                        checked="${instance.assignedLabelString!=null}" inline="true">
         <f:entry title="${%Label Expression}"
-                 description="If not set, the label of the promoted build will be used.">
+                 description="If not set, the label of the promoted build will be used." field="assignedLabelString">
```
Comment (amuniz) [inline src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly:47]: Why was this needed?

**Review by amuniz — APPROVED** [review]

File: src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly (hunk 1 above)
Comment (oleg-nenashev) [reply src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly:47]: Yes, it should not ne needed

Comment (aHenryJard) [reply src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly:47]: without the `field` the help message is not displayed at all (no ? button on the right)

Comment (jtnord) [reply src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly:47]: I wonder if this is a bug with an optional block or something - should not be needed as this should be on the textbox field below.

Comment (Vlatombe) [reply src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly:47]: IMO this is correct (the help link is determined from `f:entry`, however the textbox can be simplified now to `<f:textbox autoCompleteDelimChar=" "/>`

Comment (aHenryJard) [reply src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly:47]: ok let me change it then.

**Review by Vlatombe — APPROVED** [review]

**Comment by jglick** [comment]
https://github.com/jenkinsci/promoted-builds-plugin/releases/tag/promoted-builds-3.10
